torch_geometric/nn/conv/lightgcn_conv.py

```diff
@@ -0,0 +1,63 @@
+from torch_geometric.typing import Adj
+
+import torch
+from torch import Tensor
+from torch_geometric.nn import MessagePassing
+
+
+class LightGCNConv(MessagePassing):
+    r"""The neighbor aggregation operator from the `"LightGCN: Simplifying and
+    Powering Graph Convolution Network for Recommendation"
+    <https://arxiv.org/abs/2002.02126#>`_ paper
+
+    .. math::
+        \textbf{e}_u^{(k+1)} =
+        \sum_{i \in N_u} \frac{1}{\sqrt{|N_u|} \sqrt{|N_i|}} \textbf{e}_i^{(k)}
+
+    .. math::
+        \textbf{e}_i^{(k+1)} =
+        \sum_{i \in N_i} \frac{1}{\sqrt{|N_i|} \sqrt{|N_u|}} \textbf{e}_u^{(k)}
+
+    Args:
+        in_channels (int): Size of each input sample, or :obj:`-1` to derive
+            the size from the first input(s) to the forward method.
+        out_channels (int): Size of each output sample.
+        num_users (int): Number of users for recommendation.
+        num_items (int): Number of items to recommend.
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+    """
+    def __init__(self, in_channels: int, out_channels: int,
+                 num_users: int, num_items: int, **kwargs):
+        super(LightGCNConv, self).__init__(**kwargs)
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+
+        self.num_users = num_users
+        self.num_items = num_items
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        pass  # There are no layer parameters to learn.
+
+    def forward(self, x: Tensor, edge_index: Adj) -> Tensor:
```

Comment:
The `forward` method does not really make use of message passing functionality:
```python
def forward(self, x: Tensor, edge_index: Adj) -> Tensor:
    norm = gcn_norm(edge_index, num_nodes=x.size(0), add_self_loops=False, dtype=x.dtype)
    return self.propagate(edge_index, x=x=, norm=norm)

def message(self, x_j: Tensor, norm: Tensor) -> Tensor:
    return norm.view(-1, 1) * x_j
```

Reply:
Thank you for the suggestion! I've updated it to use `propagate` [here](https://github.com/KathyFeiyang/pytorch_geometric/blob/feat_lightgcn/torch_geometric/nn/conv/lg_conv.py#L47-L48)